### PR TITLE
[v2] ssh keypassphrase port from v3

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Resolved bug that resulted in user not being prompted for a key passphrase if it is located in the secure credential array of the ssh profile. (port from v3)

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/zosuss/src/SshBaseHandler.ts [hunk 1]
```diff
@@ -15,36 +15,29 @@ import {
...
     /**
      * The session creating from the command line arguments / profile
      */
     protected mSession: SshSession;
 
-    /**
-     * Loaded z/OS SSH profile if needed
-     * @deprecated
-     */
-    protected mSshProfile: IProfile;
```
Comment (t1m0thyj) [inline packages/zosuss/src/SshBaseHandler.ts:46]: In V2, we can't make the breaking change to remove `mSshProfile` so we should keep it (as deprecated) for now.

File: packages/zosuss/src/SshBaseHandler.ts [hunk 2]
```diff
@@ -65,26 +58,108 @@ export abstract class SshBaseHandler implements ICommandHandler {
      */
     public async process(commandParameters: IHandlerParameters) {
         this.mHandlerParams = commandParameters;
-        // Why is this here? NOTHING uses it, but I suppose an extender MIGHT be... -awharn
-        // eslint-disable-next-line deprecation/deprecation
-        this.mSshProfile = commandParameters.profiles.get("ssh", false);
```
Comment (t1m0thyj) [inline packages/zosuss/src/SshBaseHandler.ts:70]: Same comment as above - we should keep this for V2

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Looks pretty good, thanks for porting the changes @jace-roell! Left a few comments

File: packages/zosuss/src/SshBaseHandler.ts (hunk 1 above)
Comment (jace-roell) [reply packages/zosuss/src/SshBaseHandler.ts]: resolved

**Review by anaxceron — APPROVED** [review]

File: packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts [hunk 3]
```diff
@@ -366,7 +386,7 @@ export class ConnectionPropsForSessCfg {
                 let answer;
                 while (answer === undefined) {
                     const hideText = profileSchema[value]?.secure || this.secureSessCfgProps.has(value);
-                    let promptText = `${this.promptTextForValues[value]} ${serviceDescription}`;
+                    let promptText = `${this.promptTextForValues[value] ?? `Enter your ${value} for`} ${serviceDescription}`;
```
Comment (traeok) [inline packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts:389]: SonarCloud is flagging this line for nested template literals. Thoughts on making these changes to resolve the issue?

```suggestion
                    const valuePrompt = this.promptTextForValues[value] ?? `Enter your ${value} for`;
                    let promptText = `${valuePrompt} ${serviceDescription}`;
```

**Review by traeok — COMMENTED** [review]
LGTM, thanks Jace for porting this over - I left a suggestion about a SonarCloud issue.

File: packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts (hunk 3 above)
Comment (jace-roell) [reply packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts]: resolved

**Review by traeok — APPROVED** [review]

**Review by t1m0thyj — COMMENTED** [review]
According to Zowe docs, the order of precedence for SSH commands is SSH key, then user & password:
image

When testing this PR locally if I've stored values for user and password in my base profile, and have an SSH profile that defines privateKey and contains keyPassphrase in the secure array, then I get prompted for a value for keyPassphrase as expected. But if I remove user and password from my base profile and leave keyPassphrase, then issue an SSH command, I am prompted for user and password which should not be needed.

I see the same behavior in Zowe V3, so not sure if this is actually an issue, and if so whether we want to handle it separately outside of this PR? 😋

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @jace-roell!

Sorry for the false alarm - after further discussion realized that the command behaves as expected, and my test environment was tainted by a global config file.

**Review by awharn — APPROVED** [review]

**Pull request merged**